src/mame/sega/sega_beena.cpp

```diff
@@ -71,131 +102,2074 @@
+					m_requested_fiq = true;
+				}
+			}
+
+			m_pcm->reg_w(offset, data, mem_mask);
+		})).mirror(0xffc0);
+	map(0x50020000, 0x5002007f).r(FUNC(sega_9h0_0008_state::io_sensors_r)).share("io_sensor_regs").mirror(0xff80);
+	map(0x50030000, 0x500300ff).lrw32(
+		NAME([this](offs_t offset) {
+			LOG("m_io_auxiliary_regs_r @ %08x: addr=%08x\n", m_maincpu->pc(), offset * 4);
+
+			if (offset == 0x0c/4) {
+				return io_expansion_r();
+			}
+			return offset == 0xbc/4 ? io_memcache_r() : m_io_auxiliary_regs[offset];
+		}),
+		NAME([this](offs_t offset, uint32_t data, uint32_t mem_mask) {
+			LOG("m_io_auxiliary_regs_w @ %08x: addr=%08x data=%08x mask=%08x\n", m_maincpu->pc(), offset * 4, data, mem_mask);
+
+			COMBINE_DATA(&m_io_auxiliary_regs[offset]);
+			if (offset == 0xc0/4) {
+				m_memcache_events.push(data);
+			}
+		})).share("io_auxiliary_regs").mirror(0xff00);
+
+	// Realtime clock
+	map(0x60000000, 0x600000ff).rw(FUNC(sega_9h0_0008_state::rtc_r), FUNC(sega_9h0_0008_state::rtc_w)).mirror(0xff00);
+	map(0x60010000, 0x6003ffff).ram(); // Unknown
+
+	// MIDI
+	map(0x70000000, 0x7000001f).rw(FUNC(sega_9h0_0008_state::midi_reg_r), FUNC(sega_9h0_0008_state::midi_reg_w)).mirror(0xffe0);
+	map(0x70010000, 0x7003ffff).ram(); // Unknown
+
+	// Game Mask ROM
+	map(ROM_MASK_BASE, ROM_MASK_BASE + 0x7fffff).lr8(NAME([]() { return 0xff; })).mirror(0x800000);
+
+	// Game NOR Flash ROM
+	// Some games have an unreferenced handler that jumps to 0xa0ffc000;
+	map(ROM_FLASH_BASE, ROM_FLASH_BASE + 0x7fffff).lr8(NAME([]() { return 0xff; })).mirror(0x800000);
+
+	// Direct bitmap with 2-byte BGR555 colors per pixel, same as tile palettes. Scaling also applied.
+	// The bitmap size is checked by games to be `w * h <= 0x63000`, so we allocate `0x63000 * 2 = 0xc6000`.
+	map(0xc0000000, 0xc00c5fff).ram().share("bitmap");
+	// Tilemap data for 2 layers (background and foreground).
+	map(0xc00c6000, 0xc00c68ff).ram().share("tilemap_scroll_x");
+	// Work RAM for tilemap data
+	map(0xc00c6900, 0xc00c7fff).ram();
+	map(0xc00c8000, 0xc00c9fff).ram().share("tilemap_fg");
+	map(0xc00ca000, 0xc00cbfff).ram().share("tilemap_bg");
+	// Work RAM for general use + stack (usually starts at 0xc00fff80)
+	map(0xc00cc000, 0xc00fffff).ram().share("workram");
+	// Tile data for 2 layers (background and foreground) + sprites
+	map(0xc0100000, 0xc017ffff).ram().rw(FUNC(sega_9h0_0008_state::tiles_layers_r), FUNC(sega_9h0_0008_state::tiles_layers_w));
+	map(0xc0180000, 0xc01fffff).ram().rw(FUNC(sega_9h0_0008_state::tiles_sprites_r), FUNC(sega_9h0_0008_state::tiles_sprites_w));
+}
+
+void sega_9h0_0008_state::update_sensors(offs_t offset)
+{
+	// Power status
+	// - bit 1 = 0: a "low battery" screen shows up;
+	// - bit 2 = 0: game code goes into infinite loop, maybe power button pressed?
+	if (offset == 0x38/4) {
+		m_io_sensor_regs[offset] = 3;
+	}
+}
+
+uint32_t sega_9h0_0008_state::io_sensors_r(offs_t offset)
+{
+	update_sensors(offset);
+
+	return m_io_sensor_regs[offset];
+}
```

Comment:
This is a read handler with side effects.  You need to check `machine().side_effects_disabled()` to ensure debugger accesses don’t trigger unwanted side effects.

Reply:
I did hit that issue a few times, glad to have it fixed!